jme3-terrain/src/main/resources/Common/MatDefs/Terrain/GBufferPack/AdvancedPBRTerrainGBufferPack.frag

```diff
@@ -0,0 +1,516 @@
+uniform sampler2DArray m_MetallicRoughnessAoEiTextureArray;
+
+//texture-slot params for 12 unique texture slots (0-11) where the integer value points to the desired texture's index in the corresponding texture array:
+#for i=0..12 ( $0 )
+    uniform int m_AfflictionMode_$i;
+    uniform float m_Roughness_$i;
+    uniform float m_Metallic_$i;
+    uniform float m_AlbedoMap_$i_scale;
+    uniform vec4 m_EmissiveColor_$i;
+    
+    #ifdef ALBEDOMAP_$i
+        uniform int m_AlbedoMap_$i;
+    #endif        
+    #ifdef NORMALMAP_$i
+        uniform int m_NormalMap_$i;
+    #endif    
+    #ifdef METALLICROUGHNESSMAP_$i
+        uniform int m_MetallicRoughnessMap_$i;
+    #endif
+#endfor 
+
+//3 alpha maps :
+#ifdef ALPHAMAP
+  uniform sampler2D m_AlphaMap;
+#endif
+#ifdef ALPHAMAP_1
+  uniform sampler2D m_AlphaMap_1;
+#endif
+#ifdef ALPHAMAP_2
+  uniform sampler2D m_AlphaMap_2;
+#endif
+
+#ifdef DISCARD_ALPHA
+    uniform float m_AlphaDiscardThreshold;
+#endif
+
+//fog vars for basic fog :
+#ifdef USE_FOG
+#import "Common/ShaderLib/MaterialFog.glsllib"
+    uniform vec4 m_FogColor;
+    float fogDistance;
+
+    uniform vec2 m_LinearFog;
+#endif
+#ifdef FOG_EXP
+    uniform float m_ExpFog;
+#endif
+#ifdef FOG_EXPSQ
+    uniform  float m_ExpSqFog;
+#endif
+
+//sun intensity is a secondary AO value that can be painted per-vertex in the red channel of the 
+// vertex colors, or it can be set as a static value for an entire material with the StaticSunIntensity float param 
+#if defined(USE_VERTEX_COLORS_AS_SUN_INTENSITY) 
+    varying vec4 vertColors; 
+#endif
+
+#ifdef STATIC_SUN_INTENSITY
+    uniform float m_StaticSunIntensity;
+#endif
+//sun intensity AO value is only applied to the directional light, not to point lights, so it is important to track if the 
+//sun is more/less bright than the brightest point light for each fragment to determine how the light probe's ambient light should be scaled later on in light calculation code
+float brightestPointLight = 0.0;
+
+//optional affliction paramaters that use the AfflictionAlphaMap's green channel for splatting m_SplatAlbedoMap and the red channel for splatting desaturation  :
+#ifdef AFFLICTIONTEXTURE 
+    uniform sampler2D m_AfflictionAlphaMap;
+#endif
+#ifdef USE_SPLAT_NOISE
+     uniform float m_SplatNoiseVar;
+#endif
+//only defined for non-terrain geoemtries and terrains that are not positioned nor sized in correlation to the 2d array of AfflictionAlphaMaps used for splatting accross large tile based scenes in a grid
+#ifdef TILELOCATION
+    uniform float m_TileWidth;
+    uniform vec3 m_TileLocation;
+#endif
+#ifdef AFFLICTIONALBEDOMAP
+    uniform sampler2D m_SplatAlbedoMap;
+#endif
+#ifdef AFFLICTIONNORMALMAP
+    uniform sampler2D m_SplatNormalMap;
+#endif
+#ifdef AFFLICTIONROUGHNESSMETALLICMAP
+    uniform sampler2D m_SplatRoughnessMetallicMap;
+#endif
+#ifdef AFFLICTIONEMISSIVEMAP
+    uniform sampler2D m_SplatEmissiveMap;
+#endif
+
+uniform int m_AfflictionSplatScale;
```

Comment:
I discussed renaming this in the past when I submitted , and we ultimately decided to keep the naming because I don't think another better name could be used to describe the functionality since it does more than 1 thing.

The "affclition" stuff could all be removed if really needed, it mostly was just left because it doesn't hurt anything and is a bonus feature if anyone ever wants to use it. It essentially is just an extra alpha map that uses the R channel for a desaturation effect (to simulate things like grass textures turning brown/gray) and in the G channel it does texture splatting with suppor for noise to look natural with a low-res alpha map.

So I uiltimately called this unique alpha-mapping "affliction" since that encompassed the 2 features based on how they are used in my game to "afflict" the terrain with desaturation and slime effects dynamically. (and in the future I plan to add things like dynamic wetness and dynamic snowy-ness to the remaining 2 texture channels of the affliction alpha map.

So I would ultimately not advise removing the word "affliction" from anything since its an optional feature and shouldn't be confused with the normal terrain textures. Or if the affliction feature causes any issues then it would also be okay to just remove all of the affliction related params and gut the features entirely.